WikiWatershed/model-my-watershed
# MapShed Show Status

## Overview

Adds UI elements that communicate that GIS Data is being gathered for MapShed projects. Only the last two commits are relevant to this PR.

## Demo

Waiting for GIS Data:

![image](https://cloud.githubusercontent.com/assets/1430060/15522164/92850ff0-21de-11e6-83ce-3433813fada7.png)

Waiting for Results:

![image](https://cloud.githubusercontent.com/assets/1430060/15522173/a3b58930-21de-11e6-8137-3143c8a0b172.png)

Have Results:

![image](https://cloud.githubusercontent.com/assets/1430060/15522178/ab29d464-21de-11e6-84fa-272f2c097680.png)

Error State:

![image](https://cloud.githubusercontent.com/assets/1430060/15522154/7c1c0692-21de-11e6-99be-3b5793dc6725.png)

## Testing Instructions

Bundle the scripts.

 * Create a new MapShed project. You should see a spinner as GIS Data is gathered. Once the data is gathered, that screen should be replaced by the Result Type tab headers, each with spinners of their own as they wait for respective results.
 * Reload a saved MapShed project. You should not see a spinner, since GIS Data is already gathered.
 * Create a TR-55 project. You should not see a spinner, as no GIS Data needs to be gathered.
 * Select a very large area. GIS Data step should timeout, and you should see the spinner turn into an error sign.

Builds on top of #1326 
Connects #1298 
Connects #1323
